src/components/DigitalConverterComponent.cpp

```diff
@@ -0,0 +1,119 @@
+#include <iostream>
+
+#include <hyro/factory/CommandFactory.h>
+#include <hyro/DigitalConverterComponent.h> 
+
+using namespace std::chrono_literals;
+
+namespace hyro
+{
+hyro::Signal message_signal_rec;
+hyro::Signal message_output;
+Thresholding thold;
+std::shared_ptr<HyroLogger> DigitalConverterComponent::s_logger = HyroLoggerManager::CreateLogger("DigitalConverterComponent");
+
+Result
+DigitalConverterComponent::init (const ComponentConfiguration & configuration)
+{
+  /**
+   * @brief Initializes DigitalConverterComponent
+   * 
+   */
+
+  std::shared_ptr<ChannelOutput<std::vector<int>>>
+  m_dummy = this->registerOutput<std::vector<int>>("fix_dynamic"_uri, configuration);
+
+  m_input = this->registerInput<Signal>("value"_uri, configuration);
+  m_output = this->registerOutput<double>("digital"_uri, configuration);
+
+  registerDynamicProperty<double>("amplitude", 
+                                  &DigitalConverterComponent::setAmplitude,
+                                  &DigitalConverterComponent::getAmplitude,
+                                  this
+                                  );
+
+  registerDynamicProperty<double>("threshold", 
+                                  &DigitalConverterComponent::setThreshold,
+                                  &DigitalConverterComponent::getThreshold,
+                                  this
+                                  );
+
+  // Always return a result code.
+  return Result::RESULT_OK;
+}
+
+// Set threshold dynamic property
+bool 
+DigitalConverterComponent::setThreshold(double th) 
+{ 
+  threshold = th; 
+  thold.setThreshold(threshold);
+  return true;
+}
+
+// Get threshold dynamic property
+double 
+DigitalConverterComponent::getThreshold() 
+{ 
+  return threshold; 
+}
+
+// Set amplitude dynamic property
+bool 
+DigitalConverterComponent::setAmplitude(double amp) 
+{
+  amplitude = amp;
+  thold.setAmplitude(amplitude);
+  return true; 
+}
+
+// Get amplitude dynamic property
+double 
+DigitalConverterComponent::getAmplitude() 
+{ 
+  return amplitude; 
+}
+
+Result
+DigitalConverterComponent::check ()
```

Comment:
Please check if the connection was established with success here.

```c++
  if (m_input->connectionStatus() != ConnectionStatus::CONNECTION_UP)
    return Result::RESULT_ERROR;
```

Reply:
I implemented this check, it works just fine in the main code. Though it generates a failure when used in the test. Is there something else that has to be done in order to make it work in both scenarios?

```
[2019-01-31 13:57:50.240429][trace][StateMachine] [/digital] Checking -> Fault
/home/alvaro/dev/hyro/workspace/src/hyro_signal_generator/tests/test_signal_generator.cpp:51: Failure
Expected equality of these values:
  Result::RESULT_OK
    Which is: 4-byte object <00-00 00-00>
  digital_converter_state_machine.check()
    Which is: 4-byte object <02-00 00-00>
```

Reply:
Your testing is failing because you need first start the output channel that will be connected before call check function.

Probably, in the test, the channel which connects to the input channel of digital_converter was not started.